### PR TITLE
[Tests-Only] Accept all shares matching the path when accepting shares

### PR DESCRIPTION
**Pull request by aduffeck**
## Description

This pr changes the `When /^user "([^"]*)" (declines|accepts) share "([^"]*)" offered by user "([^"]*)" using the sharing API` testing step to accept all shares matching the path instead of just the first one.
This is required for test cases like [this one](https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiShareManagementToShares/mergeShare.feature#L24) where both a group and a user share for the same resource need to be considered.

## Motivation and Context

It allows for properly test the existing test cases.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [X] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 1]
```diff
@@ -2906,28 +2906,29 @@ public function userReactsToShareOfferedBy($user, $action, $share, $offeredBy, $
...
 			__METHOD__ . " could not find share $share, offered by $offeredBy to $user"
 		);
-		$url = "/apps/files_sharing/api/v{$this->sharingApiVersion}" .
-			"/shares/pending/$shareId";
-		if (\substr($action, 0, 7) === "decline") {
-			$httpRequestMethod = "DELETE";
-		} elseif (\substr($action, 0, 6) === "accept") {
-			$httpRequestMethod = "POST";
-		}
+		foreach($shareIds as $shareId) {
```
Comment (butonic) [inline tests/acceptance/features/bootstrap/Sharing.php:2916]:
```suggestion
		foreach ($shareIds as $shareId) {
```

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiWebdavLocks3-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32730/89/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiWebdavLocks3-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32735/89/1

**Comment by individual-it** [comment]
wouldn't it be enough to call the original step several times? This is done e.g. in
https://github.com/owncloud/core/blob/f379dd30e8366dfc6115689e7c69aeefd718d70e/tests/acceptance/features/apiWebdavLocks3/independentLocksShareToShares.feature#L41

`has accepted the first pending` and `has accepted the next pending` is only calls `userReactsToShareOfferedBy`

https://github.com/owncloud/core/blob/31e1891771877a561895b7c21700d69c407c059b/tests/acceptance/features/bootstrap/Sharing.php#L3025

If we need a step to accept ALL shares with that name I would prefer to have it named more explicitly e.g.
`When /^user "([^"]*)" (declines|accepts) all shares "([^"]*)" offered by user "([^"]*)" using the sharing API`
and
`When /^user "([^"]*)" (declines|accepts) share "([^"]*)" offered by user "([^"]*)" using the sharing API`

**Comment by aduffeck** [comment]
@individual-it you are right, your proposal looks like the cleaner approach. I'll try to make it work using `has accepted the first pending share` and `has accepted the next pending share` instead. Thanks :+1:

**Pull request closed without merging**